projectile.el

```diff
@@ -1017,27 +1021,31 @@ The cache is created both in memory and on the hard drive."
     (projectile-invalidate-cache nil)))
 
 ;;;###autoload
-(defun projectile-discover-projects-in-directory (directory)
+(defun projectile-discover-projects-in-directory (directory &optional depth)
   "Discover any projects in DIRECTORY and add them to the projectile cache.
-This function is not recursive and only adds projects with roots
-at the top level of DIRECTORY."
-  (interactive
-   (list (read-directory-name "Starting directory: ")))
+
+If DEPTH is non-nil recursively descend exactly DEPTH levels below DIRECTORY and
+discover projects there."
   (if (file-exists-p directory)
-      (let ((subdirs (directory-files directory t directory-files-no-dot-files-regexp t)))
-        (mapc
-         (lambda (dir)
-           (when (and (file-directory-p dir) (projectile-project-p dir))
-             (projectile-add-known-project dir)))
-         subdirs))
+      (let ((subdirs (directory-files directory t)))
```

Comment:
The indentation after the `let` looks off. I assume you wanted to nest that `dolist`.